src/main/java/com/microsoft/sqlserver/jdbc/SQLServerSpatialDatatype.java

```diff
@@ -48,13 +48,32 @@
     protected int currentFigureIndex = 0;
     protected int currentSegmentIndex = 0;
     protected int currentShapeIndex = 0;
+    protected int currentWKBPointIndex = 0;
+    protected int currentWKBFigureIndex = 0;
+    protected int currentWKBSegmentIndex = 0;
+    protected int currentWKBShapeIndex = 0;
     protected double xValues[];
     protected double yValues[];
     protected double zValues[];
     protected double mValues[];
-    protected Figure figures[];
-    protected Shape shapes[];
-    protected Segment segments[];
+    protected Figure figures[] = {};
+    protected Shape shapes[] = {};
+    protected Segment segments[] = {};
+
+    // WKB properties
+    protected byte[] wkb;
+    protected byte endian = 1; // little endian
```

Comment:
Is the returned value little endian only?

Reply:
For feature when we implement STGeomFromWKB, we should keep in mind that both values are possible when user provides WKB.

Reply:
Yes. Returned value is little endian only (as SQL Server does). I will keep this in mind when the user provides WKB.